sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -701,6 +701,57 @@ class DatasetSuite extends QueryTest
       1 -> "a", 2 -> "bc", 3 -> "d")
   }
 
+  test("SPARK-34806: observation on datasets") {
+    val namedObservation = Observation("named")
+    val unnamedObservation = Observation()
+
+    val df = spark.range(100)
+    val observed_df = df
+      .observe(
+        namedObservation,
+        min($"id").as("min_val"),
+        max($"id").as("max_val"),
+        sum($"id").as("sum_val"),
```

Comment:
one minor comment: what happens if there are duplicated names like `min(...).as("a"), max(...).as("a")`? Do we silently drop one value or do we fail at runtime?

Reply:
Yes, It behaves identical to `Row.getValuesMap(Row.schema.fieldNames)`, which drops all but the last occurrence of a column name.